youtube_transcript_api/_api.py

```diff
@@ -99,47 +99,39 @@ def get_transcript(cls, video_id, languages=None, proxies=None):
 
 class _TranscriptFetcher():
     WATCH_URL = 'https://www.youtube.com/watch?v={video_id}'
-    API_BASE_URL = 'https://www.youtube.com/api/{api_url}'
+    API_BASE_URL = 'https://www.youtube.com/api/'
     LANGUAGE_REGEX = re.compile(r'(&lang=.*&)|(&lang=.*)')
     TIMEDTEXT_STRING = 'timedtext?v='
 
     def __init__(self, video_id, languages, proxies):
         self.video_id = video_id
         self.languages = languages
         self.proxies = proxies
+        self.matched_splits = []
 
     def fetch(self):
         if self.proxies:
             fetched_site = requests.get(self.WATCH_URL.format(video_id=self.video_id), proxies=self.proxies).text
         else:
             fetched_site = requests.get(self.WATCH_URL.format(video_id=self.video_id)).text
-        timedtext_splits = fetched_site.split(self.TIMEDTEXT_STRING)
-        timedtext_url_start = (
-            timedtext_splits[2].find(self.TIMEDTEXT_STRING)
-            + len(timedtext_splits[0])
-            + len(timedtext_splits[1])
-            + len(self.TIMEDTEXT_STRING) + 1
-        )
-
-        for language in (self.languages if self.languages else [None,]):
-            response = self._execute_api_request(fetched_site, timedtext_url_start, language)
+        timedtext_splits = [split[:split.find('"')]
+                .replace('\\u0026', '&')
+                .replace('\\', '') 
+                for split in fetched_site.split(self.TIMEDTEXT_STRING)]
+        for language in (self.languages if self.languages else ['en']):
+            self.matched_splits = [split for split in timedtext_splits if '&lang={}'.format(language) in split]
+            if self.matched_splits:
+                break
+        if self.matched_splits:
+            timedtext_url = min(self.matched_splits, key=len)
```

Comment:
What's the reasoning behind taking the smallest split? Is this really reliable? 🤔

Reply:
The reasoning is that the smallest split will be the non-asr captions, as `&kind=asr` should make those splits longer.